.github/workflows/go.yml

```diff
@@ -3,6 +3,7 @@ name: Go
 on:
   push:
     branches: [ main ]
+    tags: [ v** ]
```

Comment:
Probably silly Q: releases have the form vX.Y.Z right? what do the double stars signify here?

Reply:
Semantic Versioning specifies a vX.Y.Z, where X,Y,Z are >=0  (though ironically I've never seen Z as 0 in practice)

The ** matches anything so the only restriction is the version has to start with a 'v'  - so for example it makes it straightforward to create a tag `v0.4.4-preview` or `v1.0.0-beta` etc..   Github/git doesn't enforce any syntax for the tag itself.   A tightly defined regular expression would be possible here, but balancing the more rigid and complex regex against the freeform with just enforcing the 'v' prefix

The syntax defined for matching either tags or branch names is defined at **https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet**

Github has chosen a hybrid scheme of both the standard Regular Expression syntax and GLOB patterns more usually seen for file matching. (where ** would match one or more subdirectories but * only matches one directory.  v** is not in and off itself a valid regular expression.

Pure regexp would need to be something like `v\d+\.\d+\.\d+(?:-\w+)?`  Given tags etc can be edited later if a mistake is made it's probably not worth the extra effort of getting the regex correct.

Reply:
Thanks @mbwhite !